src/libcore/tests/test_xml.py

```diff
@@ -297,3 +297,269 @@ def test20_upgrade_tree(variant_scalar_rgb):
+            <float name="radius" value="10.0"/>
+            <boolean name="flip_normals" value="false"/>
+        </shape>
+    """)
+
+    assert str(s1) == str(s2)
+    assert str(s1) == str(s3)
+    assert str(s1) == str(s4)
+
+
+def test24_dict_nested(variants_all_rgb):
+    from mitsuba.core import xml
+
+    s1 = xml.load_dict({
+        "type" : "sphere",
+        "emitter" : {
+            "type" : "area",
+        }
+    })
+
+    s2 = xml.load_string("""
+        <shape type="sphere" version="2.0.0">
+            <emitter type="area"/>
+        </shape>
+    """)
+
+    assert str(s1) == str(s2)
+
+
+def test25_dict_nested_object(variants_all_rgb):
+    from mitsuba.core import xml
+
+    bsdf = xml.load_dict({"type" : "diffuse"})
+
+    s1 = xml.load_dict({
+        "type" : "sphere",
+        "bsdf" : bsdf
+    })
+
+    s2 = xml.load_string("""
+        <shape type="sphere" version="2.0.0">
+            <bsdf type="diffuse"/>
+        </shape>
+    """)
+
+    assert str(s1.bsdf()) == str(s2.bsdf())
+
+
+def test26_dict_rgb(variants_all_rgb):
+    from mitsuba.core import xml, Color3f
+
+    e1 = xml.load_dict({
+        "type" : "point",
+        "intensity" : {
+            "type": "rgb",
+            "value" : Color3f(0.5, 0.2, 0.5)
+        }
+    })
+
+    e2 = xml.load_dict({
+        "type" : "point",
+        "intensity" : {
+            "type": "rgb",
+            "value" : [0.5, 0.2, 0.5] # list -> Color3f
+        }
+    })
+
+    e3 = xml.load_string("""
+        <emitter type="point" version="2.0.0">
+            <rgb name="intensity" value="0.5, 0.2, 0.5"/>
+        </emitter>
+    """)
+
+    assert str(e1) == str(e2)
+    assert str(e1) == str(e3)
+
+    e1 = xml.load_dict({
+        "type" : "point",
+        "intensity" : {
+            "type": "rgb",
+            "value" : 0.5 # float -> Color3f
+        }
+    })
+
+    e2 = xml.load_string("""
+        <emitter type="point" version="2.0.0">
+            <rgb name="intensity" value="0.5"/>
+        </emitter>
+    """)
+
+    assert str(e1) == str(e2)
+
+
+def test27_dict_spectrum(variant_scalar_spectral):
+    from mitsuba.core import xml
+
+    e1 = xml.load_dict({
+        "type" : "point",
+        "intensity" : {
+            "type" : "spectrum",
+            "value" : [(400, 0.1), (500, 0.2), (600, 0.4), (700, 0.1)]
+        }
+    })
+
+    e2 = xml.load_string("""
+        <emitter type="point" version="2.0.0">
+            <spectrum name="intensity" value="400:0.1 500:0.2 600:0.4 700:0.1"/>
+        </emitter>
+    """)
+
+    assert str(e1) == str(e2)
+
+    e1 = xml.load_dict({
+        "type" : "point",
+        "intensity" : {
+            "type" : "spectrum",
+            "value" : 0.44
+        }
+    })
+
+    e2 = xml.load_string("""
+        <emitter type="point" version="2.0.0">
+            <spectrum name="intensity" value="0.44"/>
+        </emitter>
+    """)
+
+    assert str(e1) == str(e2)
```

Comment:
I'm not sure if that's done indirectly here, but it would be nice to check explicitly that all the processing (pre-integration or not, within emitter, etc) is done correctly (both RGB & spectral mode, maybe mono as well).